### PR TITLE
[Doc] Update security limit for search with DLS/FLS

### PR DESCRIPTION
**Pull request by ywangd**
Shard level request cache is now generally supported after #70191 
This PR updates the security limitation to state the effect.

### CODE REVIEW

**Comment by lockewritesdocs** [comment]
@elasticmachine run elasticsearch-ci/docs

**Review by lockewritesdocs — APPROVED** [review]
Left a suggestion to reword, but LGTM otherwise.

**Comment by ywangd** [comment]
@elasticmachine update branch

File: x-pack/docs/en/security/limitations.asciidoc [hunk 1]
```diff
@@ -50,7 +50,10 @@ When a user's role enables document or <<field-level-security,field level securi
 * The user cannot perform write operations:
 ** The update API isn't supported.
 ** Update requests included in bulk requests aren't supported.
-* The request cache is disabled for search requests.
+* The request cache is disabled for search requests if:
```
Comment (tvernum) [inline x-pack/docs/en/security/limitations.asciidoc:53]:
```suggestion
* The request cache is disabled for search requests if either of the following are true:
```

**Review by tvernum — APPROVED** [review]

**Comment by ywangd** [comment]
@elasticmachine update branch

**Pull request merged**